gdk/common/recipe_file/CaseInsensitiveRecipeFile.py

```diff
@@ -0,0 +1,92 @@
+import json
+import yaml
+from pathlib import Path
+from requests.structures import CaseInsensitiveDict
+
+
+class CaseInsensitiveRecipeFile:
+    def write(self, file_path: Path, content: CaseInsensitiveDict) -> None:
+        """
+        Writes CaseInsensitiveDict contents to a JSON or a YAML file based on the file path.
+        """
+        self._write(file_path, self._to_dict(content))
+
+    def read(self, file_path: Path) -> CaseInsensitiveDict:
+        """
+        Reads a JSON or a YAMl file contents as a CaseInsensitiveDict and returns it.
+        """
+        return self._from_dict(self._read(file_path))
+
+    def _write(self, file_path, content):
```

Comment:
This method is only called by write, having it as a separate method is not necessary (at least for now)

Reply:
Agreed. Without `_write()` and `_read()`, we'll have to convert to_dict or from_dict for both json and yaml operations. Just doing that only one time with this additional _write and _read.